src/cloud-api-adaptor/test/e2e/assessment_helpers.go

```diff
@@ -218,6 +189,55 @@ func ComparePodLogString(ctx context.Context, client klient.Client, customPod v1
 	return podLogString, nil
 }
 
+// Note: there are currently two event types: Normal and Warning, so Warning includes errors
+func GetPodEventWarningDescriptions(ctx context.Context, client klient.Client, pod v1.Pod) (string, error) {
+	clientset, err := kubernetes.NewForConfig(client.RESTConfig())
+	if err != nil {
+		return "", err
+	}
+
+	events, err := clientset.CoreV1().Events(pod.Namespace).List(ctx, metav1.ListOptions{FieldSelector: fmt.Sprintf("involvedObject.name=%s", pod.Name)})
+	if err != nil {
+		return "", err
+	}
+
+	var descriptionsBuilder strings.Builder
+	for _, event := range events.Items {
+		if event.Type == v1.EventTypeWarning {
+			descriptionsBuilder.WriteString(event.Message)
+		}
+	}
+	return descriptionsBuilder.String(), nil
+}
+
+func ComparePodEventWarningDescriptions(ctx context.Context, t *testing.T, client klient.Client, pod v1.Pod, expectedPodEventDescribe string) error {
+	retries := 1
+	delay := 10 * time.Second
+
+	if pod.Status.Phase != v1.PodFailed {
+		// If not failed state we might have to wait/retry until the error happens
+		retries = int(WAIT_POD_RUNNING_TIMEOUT / delay)
+	}
+
+	var err error = nil
+	for retries > 0 {
+		podEventsDescriptions, podErr := getStringFromPod(ctx, client, pod, GetPodEventWarningDescriptions)
+		if podErr != nil {
+			return podErr
+		}
+
+		t.Logf("podEvents: %+v\n", podEventsDescriptions)
```

Comment:
Hi @stevenhorsman !

What does the "+" in "+v" do? I vaguely remember I had to remove it to print the string correctly, I just don't remember where.

Reply:
The `%+v` includes the field names of the struct. In this case it's a string IIRC, so I should just switch to `%s` instead. I think the `%+v` was from an earlier version. Sorry about that.